src/frequenz/sdk/actor/_run_utils.py

```diff
@@ -38,18 +38,22 @@ async def run(*actors: Actor) -> None:
         done_tasks, pending_tasks = await asyncio.wait(
             pending_tasks, return_when=asyncio.FIRST_COMPLETED
         )
-
         # This should always be only one task, but we handle many for extra safety
         for task in done_tasks:
-            # Cancellation needs to be checked first, otherwise the other methods
-            # could raise a CancelledError
-            if task.cancelled():
+            # BackgroundService returns a BaseExceptionGroup containing multiple
+            # exceptions. The 'await' statement raises these exceptions, and 'except*'
+            # is used to handle them as a group. If the task raises multiple different
+            # exceptions, 'except*' will be invoked multiple times, once for each
+            # exception group.
+            try:
+                await task
+            except* asyncio.CancelledError:
                 _logger.info("Actor %s: Cancelled while running.", task.get_name())
-            elif exception := task.exception():
+            except* BaseException as err:  # pylint: disable=broad-exception-caught
```

Comment:
This would be a change of behaviour, but I think we should only catch `Exception` here, we probably don't want to catch stuff like [`KeyboardInterrupt`](https://docs.python.org/3/library/exceptions.html#KeyboardInterrupt), [`GeneratorExit`](https://docs.python.org/3/library/exceptions.html#GeneratorExit) or [`SystemExit`](https://docs.python.org/3/library/exceptions.html#SystemExit). `CancelledError` is probably the only [`BaseException`](https://docs.python.org/3/library/exceptions.html#BaseException) we ever want to catch.

Reply:
I also think so, but we had dedicated tests for BaseException.
Removed in last commit.